docs/architecture/adr-002-evm-hooks.md

```diff
@@ -0,0 +1,52 @@
+# ADR 002: EVM Hooks
+
+## Changelog
+
+- 2021-08-11: first draft
+
+## Status
+
+PROPOSED
+
+## Abstract
+
+The current ADR proposes a hook interface to EVM module, with the goal of extend the functionality externally,
+specifically to support communication from EVM contract to native modules through logs.
+
+## Context
+
+There are some requirements to enable EVM contract to communicate with cosmos native modules, which could have multiple
+use cases. We need some kind of hooks to support these extension requirements in a generic way.
+
+## Decision
+
+This ADR propose to add `EvmHooks` interface:
+
+```golang
+type EvmHooks interface {
+	  PostTxProcessing(ctx sdk.Context, txHash ethcmn.Hash, logs []*ethtypes.Log) error
+}
+```
+
+- `PostTxProcessing` is called after EVM transaction finished, executed with the same cache context as the EVM
+  transaction execution, if `PostTxProcessing` returns an error, the whole EVM transaction is reverted.
```

Comment:
explain why are you reverting the entire tx vs just doing a no-op for the hook post tx processing

Reply:
added.
it makes the implementor to decide if certain api want to revert on error or not.